spikeextractors/extractors/neoextractors/neobaseextractor.py

```diff
@@ -83,8 +83,9 @@ def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
         scaled_traces = self.neo_reader.rescale_signal_raw_to_float(raw_traces, dtype='float32',
                                                                     channel_indexes=None, channel_names=None,
                                                                     channel_ids=channel_ids)
+        channel_idxs = np.array([list(channel_ids).index(ch) for ch in channel_ids])
         # and then to uV
-        scaled_traces *= self.additional_gain
+        scaled_traces *= self.additional_gain[0, channel_idxs]
```

Comment:
The ```[0, ``` is wrong this first dimention is needed to force the broacasting when there is ambiguity on dims.
So ```[:, ```

And to speed up this I would put 
```
If channel_ids is None:
    additional_gain = self.additional_gain
else:
    channel_idxs = np.array([list(channel_ids).index(ch) for ch in channel_ids])
    additional_gain = self.additional_gain[:, channel_idxs]
scaled_traces *= additional_gain[:, channel_idxs]
```

Reply:
Channel IDs is never none due to the decoratorm you can push the changes you'd like, but additional_gain has a shape (1, num_channels)

Reply:
I think it is a bad idea that channel_ids is never None.

Because when you need slice(None) to get a full dimenssion is really really faster that doing
array.take(...).

I don't known when this was done but forcing np.arange(nb_channel) instead of slice(None) is really really a perf killer to fetch data on a memap for instance.

Reply:
no extractors slice except Biocam, and in that case the slicing is done in place (if channel ids are ordered and with no gaps)